tests/robust/robust_fixtures.py

```diff
@@ -0,0 +1,208 @@
+        )
+        scale = pyro.sample(
+            "covariate_scale", dist.LogNormal(0, 1).expand((self.p,)).to_event(1)
+        )
+        return loc, scale
+
+    def forward(self, N: int = 1):
+        intercept = self.sample_intercept()
+        outcome_weights = self.sample_outcome_weights()
+        propensity_weights = self.sample_propensity_weights()
+        tau = self.sample_treatment_weight()
+        x_loc, x_scale = self.sample_covariate_loc_scale()
+        with pyro.plate("obs", N, dim=-1):
+            X = pyro.sample("X", dist.Normal(x_loc, x_scale).to_event(1))
+            A = pyro.sample(
+                "A",
+                dist.Bernoulli(
+                    logits=torch.einsum("...np,...p->...n", X, propensity_weights)
+                ),
+            )
+            return pyro.sample(
+                "Y",
+                self.link_fn(
+                    torch.einsum("...np,...p->...n", X, outcome_weights)
+                    + A * tau
+                    + intercept
+                ),
+            )
+
+
+class KnownCovariateDistModel(HighDimLinearModel):
+    def sample_covariate_loc_scale(self):
+        return torch.zeros(self.p), torch.ones(self.p)
+
+
+class BenchmarkLinearModel(HighDimLinearModel):
+    def __init__(
+        self,
+        p: int,
+        link_fn: Callable[..., dist.Distribution],
+        alpha: int,
+        beta: int,
+        treatment_weight: float = 0.0,
+    ):
+        super().__init__(p, link_fn)
+        self.alpha = alpha  # sparsity of propensity weights
+        self.beta = beta  # sparisty of outcome weights
+        self.treatment_weight = treatment_weight
+
+    def sample_outcome_weights(self):
+        outcome_weights = 1 / math.sqrt(self.beta) * torch.ones(self.p)
+        outcome_weights[self.beta :] = 0.0
+        return outcome_weights
+
+    def sample_treatment_null_weight(self):
+        return torch.tensor(0.0)
+
+    def sample_propensity_weights(self):
+        propensity_weights = 1 / math.sqrt(self.alpha) * torch.ones(self.p)
+        propensity_weights[self.alpha :] = 0.0
+        return propensity_weights
+
+    def sample_treatment_weight(self):
+        return torch.tensor(self.treatment_weight)
+
+    def sample_intercept(self):
+        return torch.tensor(0.0)
+
+    def sample_covariate_loc_scale(self):
+        return torch.zeros(self.p), torch.ones(self.p)
+
+
+class MLEGuide(torch.nn.Module):
+    def __init__(self, mle_est: ParamDict):
+        super().__init__()
+        self.names = list(mle_est.keys())
+        for name, value in mle_est.items():
+            setattr(self, name + "_param", torch.nn.Parameter(value))
+
+    def forward(self, *args, **kwargs):
+        for name in self.names:
+            value = getattr(self, name + "_param")
+            pyro.sample(name, dist.Delta(value))
+
+
+def closed_form_ate_correction(X_test: Point[T], theta: ParamDict):
```

Comment:
nit: I think these types should be `typing.TypedDict`s since you're expecting specific names:
```
class ATETestPoint(TypedDict):
    X: torch.Tensor
    A: torch.Tensor
    Y: torch.Tensor


class ATEParamDict(TypedDict):
    propensity_weights: torch.Tensor
    outcome_weights: torch.Tensor
    treatment_weight: torch.Tensor
    intercept: torch.Tensor


def  closed_form_ate_correction(X_test: ATETestPoint, theta: ATEParamDict) -> Tuple[torch.Tensor, torch.Tensor]:
    ...
```

Reply:
Got it thanks!